### PR TITLE
fix(e2e): Rename files

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

## Description of the changes
Added files only used in E2E tests to the E2E project. If this needs to be shared in the future, we can create a common project and add a project reference to it. Adding links to files is not the best option and makes it hard to do things like refactoring from  VS. Also keeping it in a common place like that might lead us to accidentally take dependency on something that just belongs to the tests.

Renamed the following:

WriteLine -> Trace
TestLogging -> TestLogger
VerboseTestLogging -> VerboseTestLogger
EventSourceTestLogging -> EventSourceTestLogger

Moved the following to E2E project:
Following files were added as a compile link in the .csproj in the E2E project and not used in any other project.

Common\CodeAnalysisOverrides.cs
Common\Configuration.cs
Common\Configuration.IoTHub.cs
Common\Configuration.Provisioning.cs
Common\Configuration.Provisioning.cs
Common\Configuration.Storage.cs
Common\Configuration.AzureSecurityCenterForIoTLogAnalytics.cs
Common\EventSourceTestLogging.cs
Common\TestModule.cs
Common\TestLogging.cs
Common\TestDevice.cs
Common\VerboseTestLogging.cs
Common\CustomWebProxy.cs
Common\ConsoleEventListener.cs

### CODE REVIEW

File: e2e/test/E2ETests.csproj [hunk 1]
```diff
@@ -14,49 +14,6 @@
     <CommonTest>$(RootDir)\common\test</CommonTest>
```
Comment (abhipsaMisra) [inline e2e/test/E2ETests.csproj:14]: this can now be removed

File: e2e/test/Common/CodeAnalysisOverrides.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+
+// Required by Configuration.*
+[assembly: SuppressMessage("Microsoft.Design", "CA1724:TypeNamesShouldNotMatchNamespaces")]
```
Comment (abhipsaMisra) [inline e2e/test/Common/CodeAnalysisOverrides.cs:7]: do you know which classes are using this?

File: e2e/test/Common/ConsoleEventListener.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using System.Linq;
+
+namespace System.Diagnostics.Tracing
+{
+    public sealed class ConsoleEventListener : EventListener
```
Comment (abhipsaMisra) [inline e2e/test/Common/ConsoleEventListener.cs:9]: FYI - this class is referenced in our readme here: https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/CaptureLogs#console-logging
Make sure this path is updated as well

File: e2e/test/Common/EventSourceTestLogger.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    // f7ac322b-77f1-5a2d-0b56-ec79a41e82a2
+    [EventSource(Name = "Microsoft-Azure-Devices-TestLogging")]
+    internal class EventSourceTestLogger : EventSource
```
Comment (abhipsaMisra) [inline e2e/test/Common/EventSourceTestLogger.cs:10]: don't unit test files use this class as well?

File: e2e/test/Common/CodeAnalysisOverrides.cs (hunk 2 above)
Comment (vinagesh) [reply e2e/test/Common/CodeAnalysisOverrides.cs:7]: I have not checked. I just copied everything the project was referring to.

File: e2e/test/Common/ConsoleEventListener.cs (hunk 3 above)
Comment (vinagesh) [reply e2e/test/Common/ConsoleEventListener.cs:9]: Thanks for the reference. I'll update it.

File: e2e/test/Common/EventSourceTestLogger.cs (hunk 4 above)
Comment (vinagesh) [reply e2e/test/Common/EventSourceTestLogger.cs:10]: Everything builds after removing it. Plus I looked at all the .csproj for the links and did not find anything.

**Pull request merged**